SongBird/app/src/main/java/shuvalov/nikita/songbird/MainActivity.java

```diff
@@ -0,0 +1,165 @@
+package shuvalov.nikita.songbird;
+
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.AsyncTask;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+
+import com.google.gson.Gson;
+
+import org.apache.commons.codec.binary.Base64;
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.FormBody;
+import okhttp3.Headers;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+
+public class MainActivity extends AppCompatActivity {
+
+    RecyclerView mRecyclerView;
+    Button mSearchButton;
+    EditText mUserInput;
+
+    BearerTokenResponse mBearerTokenHolder;
+    BearerTokenTask mBTtask;
+
+    ArrayList<TweetInfo> mTweetInfo;
+
+    TweetRecyclerAdapater mAdapter;
+    RecyclerView mRecyclerview;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mRecyclerView = (RecyclerView)findViewById(R.id.recycler_view);
+        mSearchButton = (Button) findViewById(R.id.enter_search);
+        mUserInput = (EditText) findViewById(R.id.user_prompt);
+
+        mRecyclerView = (RecyclerView)findViewById(R.id.recycler_view);
+
+        mTweetInfo = new ArrayList<>();
+
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
+        mAdapter = new TweetRecyclerAdapater(mTweetInfo);
+
+        mRecyclerView.setAdapter(mAdapter);
+        mRecyclerView.setLayoutManager(linearLayoutManager);
+
+
+        ConnectivityManager connectivityManager = (ConnectivityManager)getSystemService(CONNECTIVITY_SERVICE);
+        NetworkInfo info = connectivityManager.getActiveNetworkInfo();
+        String concatentatedKeySecret = AppData.CONSUMER_KEY+":"+AppData.CONSUMER_SECRET;
+        String encodedConsumerKey = new String(Base64.encodeBase64(concatentatedKeySecret.getBytes()));
+
+        if (info!=null && info.isConnected()){
+            mBTtask= new BearerTokenTask();
+            mBTtask.execute(encodedConsumerKey);
+        }else{
+            Toast.makeText(this, "No Internet Connectivity", Toast.LENGTH_SHORT).show();
+        }
+
+        mSearchButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (mBTtask!= null && mBTtask.getStatus() == AsyncTask.Status.FINISHED){
```

Comment:
Nice idea to make sure you've finished getting the access token before launching a new request!